src/Helpers/DataResolver.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace Firesphere\SolrSearch\Helpers;
+
+use Exception;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\FieldType\DBField;
+use SilverStripe\ORM\SS_List;
+use SilverStripe\View\ArrayData;
+
+class DataResolver
+{
+    protected $component;
+    protected $columns = [];
+    protected $columnName = '';
+
+    public function __construct($component, $columns = [])
+    {
+        if (!is_array($columns)) {
+            $columns = array_filter(explode('.', $columns));
+        }
+        $this->columns = $columns;
+        $this->component = $component;
+        $this->columnName = $this->columns ? array_shift($this->columns) : null;
+    }
+
+    /**
+     * @param DataObject|ArrayData|SS_List $obj
+     * @param array|string $columns
+     *
+     * @return mixed
+     * @throws Exception
+     */
+    public static function identify($obj, $columns = [])
+    {
+        $instance = new self($obj, $columns);
+        if ($obj instanceof DataObject) {
+            return $instance->resolveDataObject();
+        }
+        if ($obj instanceof ArrayData) {
+            return $instance->resolveArrayData();
+        }
+        if ($obj instanceof SS_List) {
+            return $instance->resolveList();
+        }
+        if ($obj instanceof DBField) {
+            return $instance->resolveField();
+        }
+        throw new Exception(sprintf('Class: %s, is not supported.', get_class($obj)));
```

Comment:
I think ClassInfo::getShortName($obj); would be more appropriate here?

Reply:
Cool, I like this :+1: Will implement